### PR TITLE
Migrate LDAP images to CentOS 7

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

File: testing/centos7-oj8-openldap/Dockerfile [hunk 1]
```diff
@@ -10,27 +10,30 @@
...
 COPY ./files /
 
 # CONFIGURE OPENLDAP SERVER
-RUN service slapd start && \
+RUN supervisord -c /etc/supervisord.conf && \
+    sleep 15 && \
     ldapmodify -Y EXTERNAL -H ldapi:/// -f /etc/openldap/setup/modify_server.ldif && \
     ldapmodify -Y EXTERNAL -H ldapi:/// -f /etc/openldap/setup/ldap_disable_bind_anon.ldif && \
     ldapadd -Y EXTERNAL -H ldapi:/// -f /etc/openldap/setup/memberof.ldif && \
+    ldapadd -Y EXTERNAL -H ldapi:/// -f /etc/openldap/schema/cosine.ldif && \
```
Comment (wendigo) [inline testing/centos7-oj8-openldap/Dockerfile:29]: this is required in order to have "interorgperson" object class which we are using in trino

File: testing/centos7-oj8-openldap/Dockerfile [hunk 2]
```diff
@@ -10,27 +10,30 @@
...
+    ldapadd -Y EXTERNAL -H ldapi:/// -f /etc/openldap/schema/inetorgperson.ldif && \
     ldapadd -Y EXTERNAL -H ldapi:/// -f /etc/openldap/setup/refint.ldif && \
     ldapadd -f /etc/openldap/setup/createOU.ldif -D cn=admin,dc=trino,dc=testldap,dc=com -w admin
 
 # Generate a keystore.
 RUN keytool -genkey -alias coordinator -storepass testldap -keystore /etc/openldap/certs/coordinator.jks \
-    -keypass testldap -keyalg RSA -sigalg SHA1withRSA -dname "CN=trino-master, OU=, O=, L=, S=, C=" -validity 100000
+    -keypass testldap -keyalg RSA -sigalg SHA1withRSA -dname "CN=presto-master, OU=, O=, L=, S=, C=" -validity 100000
```
Comment (wendigo) [inline testing/centos7-oj8-openldap/Dockerfile:36]: I had to revert this change in order to make Trino work with the older node names

File: testing/centos7-oj8-openldap/Dockerfile [hunk 3]
```diff
@@ -10,27 +10,30 @@
...
     # Cleanup
     && yum -y clean all && rm -rf /tmp/* /var/tmp/*
 
 # COPY CONFIGURATION
 COPY ./files /
 
 # CONFIGURE OPENLDAP SERVER
-RUN service slapd start && \
+RUN supervisord -c /etc/supervisord.conf && \
+    sleep 15 && \
```
Comment (findepi) [inline testing/centos7-oj8-openldap/Dockerfile:25]: y?

File: testing/centos7-oj8-openldap-referrals/Dockerfile [hunk 4]
```diff
@@ -10,13 +10,14 @@
...
-FROM testing/centos6-oj8-openldap:unlabelled
+FROM testing/centos7-oj8-openldap:unlabelled
 
 # COPY CONFIGURATION
 COPY ./files /
 
 # CONFIGURE OPENLDAP SERVER
-RUN service slapd start && \
+RUN supervisord -c /etc/supervisord.conf && \
+    sleep 15 && \
```
Comment (findepi) [inline testing/centos7-oj8-openldap-referrals/Dockerfile:20]: what is this?

File: testing/centos7-oj8-openldap-referrals/Dockerfile [hunk 5]
```diff
@@ -10,13 +10,14 @@
...
     ldapadd -f /etc/openldap/setup/createReferrals.ldif -D cn=admin,dc=trino,dc=testldap,dc=com -w admin
 
-CMD supervisord -c /etc/supervisord.conf
+CMD supervisord -n -c /etc/supervisord.conf
```
Comment (findepi) [inline testing/centos7-oj8-openldap-referrals/Dockerfile:23]: what's `-n`?

File: testing/centos7-oj8-openldap-referrals/Dockerfile [hunk 6]
```diff
@@ -10,13 +10,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM testing/centos6-oj8-openldap:unlabelled
+FROM testing/centos7-oj8-openldap:unlabelled
```
Comment (findepi) [inline testing/centos7-oj8-openldap-referrals/Dockerfile:13]: Looks like you could squash the commits without big damage to history (git should still recognize renames, but you may want to verify that)

File: testing/centos7-oj8-openldap/files/etc/openldap/setup/memberof.ldif [hunk 7]
```diff
@@ -1,11 +1,11 @@
-# This overlay helps in determing which group a given user belongs to
+# This overlay helps in determining which group a given user belongs to
 dn: cn=module,cn=config
 cn: module
 objectClass: olcModuleList
 olcModuleLoad: memberof.la
 olcModulePath: /usr/lib64/openldap
 
-dn: olcOverlay={0}memberof,olcDatabase={2}bdb,cn=config
+dn: olcOverlay={0}memberof,olcDatabase={2}hdb,cn=config
```
Comment (findepi) [inline testing/centos7-oj8-openldap/files/etc/openldap/setup/memberof.ldif:8]: sadly, i have no idea what this change means

cc @sopel39 @MiguelWeezardo

File: testing/centos7-oj8-openldap/files/etc/supervisord.conf [hunk 8]
```diff
@@ -4,7 +4,6 @@ logfile_maxbytes = 50MB
 logfile_backups=10
 loglevel = info
 pidfile = /var/run/supervisord.pid
-nodaemon = true
```
Comment (findepi) [inline testing/centos7-oj8-openldap/files/etc/supervisord.conf:7]: why?
it's very docker-wise when supervisor runs in the foreground

File: testing/centos7-oj8-openldap/Dockerfile (hunk 3 above)
Comment (wendigo) [reply testing/centos7-oj8-openldap/Dockerfile:25]: `supervisord -c` exits as it's running in background so it takes a couple of seconds for ldap server to get up

File: testing/centos7-oj8-openldap-referrals/Dockerfile (hunk 4 above)
Comment (wendigo) [reply testing/centos7-oj8-openldap-referrals/Dockerfile:20]: CentOS 7 image is based on systemd which is hard to use inside a docker container. This is actually a hack to get it running

File: testing/centos7-oj8-openldap-referrals/Dockerfile (hunk 5 above)
Comment (wendigo) [reply testing/centos7-oj8-openldap-referrals/Dockerfile:23]: run in foreground

File: testing/centos7-oj8-openldap/files/etc/supervisord.conf (hunk 8 above)
Comment (wendigo) [reply testing/centos7-oj8-openldap/files/etc/supervisord.conf:7]: this is why I've added `-n` instead

File: testing/centos7-oj8-openldap/files/etc/openldap/setup/memberof.ldif (hunk 7 above)
Comment (wendigo) [reply testing/centos7-oj8-openldap/files/etc/openldap/setup/memberof.ldif:8]: this is a name of the backend (database)

File: testing/centos7-oj8-openldap/Dockerfile (hunk 2 above)
Comment (MiguelWeezardo) [reply testing/centos7-oj8-openldap/Dockerfile:36]: Please put this in a comment in the file

File: testing/centos7-oj8-openldap/Dockerfile (hunk 1 above)
Comment (MiguelWeezardo) [reply testing/centos7-oj8-openldap/Dockerfile:29]: Please put this in a comment in the file

**Comment by wendigo** [comment]
Build failures for CDH and HDP images are expected. These will be fixed with https://github.com/trinodb/docker-images/pull/92

**Comment by losipiuk** [comment]
Please provide motivation in commit message

File: testing/centos7-oj8-openldap-referrals/Dockerfile (hunk 4 above)
Comment (losipiuk) [reply testing/centos7-oj8-openldap-referrals/Dockerfile:22]: Maybe run `ldapadd -f /etc/openldap/setup/createReferrals.ldif -D cn=admin,dc=trino,dc=testldap,dc=com -w admin` in a loop instead `sleep`

**Comment by wendigo** [comment]
@losipiuk done

File: testing/centos7-oj8-openldap-referrals/Dockerfile (hunk 5 above)
Comment (losipiuk) [reply testing/centos7-oj8-openldap-referrals/Dockerfile:25]: Does it kill previous running `supervisord`? Or just creates a foreground process which attaches to one you started already?

File: testing/centos7-oj8-openldap/Dockerfile [hunk 9]
```diff
@@ -10,27 +10,34 @@
...
 # CONFIGURE OPENLDAP SERVER
-RUN service slapd start && \
+# Run supervisord in background
+RUN supervisord -c /etc/supervisord.conf && \
+    # Wait for ldapd to start
+    sleep 15 && \
     ldapmodify -Y EXTERNAL -H ldapi:/// -f /etc/openldap/setup/modify_server.ldif && \
     ldapmodify -Y EXTERNAL -H ldapi:/// -f /etc/openldap/setup/ldap_disable_bind_anon.ldif && \
     ldapadd -Y EXTERNAL -H ldapi:/// -f /etc/openldap/setup/memberof.ldif && \
+    # Next two ldifs are required in order to have inetOrgPerson object class
```
Comment (losipiuk) [inline testing/centos7-oj8-openldap/Dockerfile:31]: Not sure if we need this comment. It is no harm. But it is not that these two lines are less obvious than the rest here :)

Comment (losipiuk) [reply testing/centos7-oj8-openldap/Dockerfile:31]: Yeah - I see you added this comment following review comment :)

File: testing/centos7-oj8-openldap/files/etc/openldap/setup/refint.ldif [hunk 10]
```diff
@@ -6,10 +6,10 @@ objectclass: olcModuleList
...
 
-dn: olcOverlay={1}refint,olcDatabase={2}bdb,cn=config
+dn: olcOverlay={1}refint,olcDatabase={2}hdb,cn=config
 objectClass: olcConfig
 objectClass: olcOverlayConfig
 objectClass: olcRefintConfig
 objectClass: top
 olcOverlay: {1}refint
-olcRefintAttribute: memberof member manager owner
+olcRefintAttribute: memberof member manager owner
```
Comment (losipiuk) [inline testing/centos7-oj8-openldap/files/etc/openldap/setup/refint.ldif:15]: revert

**Review by losipiuk — APPROVED** [review]

File: testing/centos7-oj8-openldap/Dockerfile (hunk 9 above)
Comment (wendigo) [reply testing/centos7-oj8-openldap/Dockerfile:31]: These are the files that are distributed with openldap itself. Previously this schema was applied by default in CentOS 6. In CentOS 7 I had to apply it by hand so that's the reason for comment (it took me some time to understand that)

File: testing/centos7-oj8-openldap/files/etc/openldap/setup/refint.ldif (hunk 10 above)
Comment (wendigo) [reply testing/centos7-oj8-openldap/files/etc/openldap/setup/refint.ldif]: done

**Review by losipiuk — APPROVED** [review]
